zgw/zaakdetails/src/test/kotlin/com/ritense/zaakdetails/documentobjectenapisync/DocumentObjectenApiSyncServiceIntTest.kt

```diff
@@ -69,43 +113,179 @@ class DocumentObjectenApiSyncServiceIntTest : BaseIntegrationTest() {
         }
 
         assertTrue { result.errors().isEmpty() }
+
+        verify(zakenApiClient, times(1)).createZaak(any(), any(), any())
+        verify(objectenApiClient, times(1)).getObjectsByObjecttypeUrlWithSearchParams(any(), any(), any(), any(), any(), any(), any())
         verify(objectenApiClient, times(1)).createObject(any(), any(), any())
-        verify(objectenApiClient, times(0)).objectUpdate(any(), any(), any())
+        verifyNoMoreInteractions(objectenApiClient)
+        verify(zakenApiClient, times(1)).createZaakObject(any(), any(), any())
     }
 
     @Test
-    fun `should sync modified document to Objecten API`() {
-        documentObjectenApiSyncService.saveSyncConfiguration(
+    fun `should update zaakdetails object and link to the zaak when zaak exists`() {
```

Comment:
I think we don't have a test for the case where we already have an zaak details object but it wasn't linked before?